cfgov/ask_cfpb/scripts/migrate_tagging.py

```diff
@@ -0,0 +1,150 @@
+                return
+            with open(csv_file, 'r') as f:
+                reader = csv.DictReader(f)
+                data = [
+                    row for row in reader
+                    if row['PrimaryTopicTag'] in SLUGS[slug]
+                ]
+            output = transform_csv(data, portal_name)
+            if not output:
+                print("No output produced")
+                return
+            output_json(output, slug)
+            print("Output {}.json".format(slug))
+        apply_tags(output)
+        print("Tag migrations took {}".format(
+            datetime.datetime.now() - starter))
+
+
+def update_feedback_default(stream_value):
+    new_stream_data = stream_value.stream_data
+    new_stream_data[0]['value'].update(
+        {'was_it_helpful_text': 'Was this answer helpful to you?'})
+    stream_value.stream_data = new_stream_data
+    return stream_value
+
+
+def apply_tags(data):
+    """Apply tagging values to answer pages."""
+    category_map = {obj.pk: obj for obj in PortalCategory.objects.all()}
+    migration_user_pk = os.getenv('MIGRATION_USER_PK')
+    user = User.objects.filter(id=migration_user_pk).first()  # default is None
+    for entry in data:
+        pages = AnswerPage.objects.filter(
+            answer_base__pk=entry.get('ask_id'))
+        for page in pages:
+            initial_status = page.status_string
+            page.get_latest_revision().publish()
+            page.user_feedback = update_feedback_default(page.user_feedback)
+            portal_ids = entry.get('portal_ids')
+            category_ids = entry.get('see_all_ids')
+            for portal_id in portal_ids:
+                topic = PortalTopic.objects.get(pk=portal_id)
+                page.portal_topic.add(topic)
+            for category_id in category_ids:
+                category_object = category_map.get(category_id)
+                if category_object:
+                    page.portal_category.add(category_object)
+            page.save_revision(user=user).publish()
+            if initial_status == 'draft':
+                page.unpublish()
+
+
+def output_json(data, slug):
+    with open("{}/{}.json".format(FIXTURE_DIR, slug), 'w') as f:
+        f.write(json.dumps(data, indent=4))
+
+
+def transform_csv(dict_rows, portal):
+    """Turn a CE spreadsheet into a json mapping file."""
+    output = []
+    portal_pk = PORTAL_IDS.get(portal)
+    for row in dict_rows:
+        categories = [
+            tag.strip() for tag in [
+                row['PrimaryCategoryTag'],
+                row['CategoryTag2'],
+                row['CategoryTag3'],
+                row['CategoryTag4']]
+            if tag.strip()]
+        category_pks = [CATEGORY_IDS.get(tag) for tag in categories
+                        if CATEGORY_IDS.get(tag)]
+        entry = {
+            'ask_id': row.get('ask_id'),
+            'portal_ids': [portal_pk],
+            'see_all_ids': category_pks,
```

Comment:
This is very minor but I find the use of `see_all_ids` a little confusing here.  We've been referring to these two fields are portal topics and categories. They do have page corollaries but given that you use `CATEGORY_IDS` above it's a little inconsistent.  I know it's a one-time script, though, so it doesn't matter too much.

Reply:
You hit the nail. The see-all naming was ill-conceived, but the json taggin files used it, and since this script is disposable, I didn't think it was worth redoing the 12 json files.